libs/utils/utils.js

```diff
@@ -1270,4 +1270,32 @@ export function loadLana(options = {}) {
   window.addEventListener('unhandledrejection', lanaError);
 }
 
+const defaultAllowedOrigins = [
+  'https://www.adobe.com',
+  'https://business.adobe.com',
+  'https://blog.adobe.com',
+  'https://milo.adobe.com',
+];
+
+let federatedContentRoot;
+export const getFederatedContentRoot = () => {
+  const { allowedOrigins = [] } = getConfig();
+  if (federatedContentRoot) return federatedContentRoot;
+
+  const { origin } = window.location;
+
+  federatedContentRoot = [...allowedOrigins, ...defaultAllowedOrigins].some((o) => origin.replace('.stage', '') === o)
+    ? origin
+    : 'https://www.adobe.com';
+
+  if (origin.includes('localhost') || origin.includes('.hlx.')) {
+    // Akamai as proxy to avoid 401s, given AEM-EDS MS auth cross project limitations
+    federatedContentRoot = origin.includes('.hlx.live')
+      ? 'https://main--federal--adobecom.hlx.live'
+      : 'https://www.stage.adobe.com';
+  }
+
+  return federatedContentRoot;
+};
+
```

Comment:
IMO we should have a dynamic import for the whole federated content or move it to its own separate module.

I would guess that most pages do **not** use federal content and thus shouldn't load it at all. The blocks that do, should load it themselves